docs/_basic/authenticating_oauth.md

```diff
@@ -29,10 +29,11 @@ const app = new App({
   installationStore: {
     storeInstallation: async (installation) => {
       // change the line below so it saves to your database
-      if (installation.isEnterpriseInstall) {
+      if (installation.isEnterpriseInstall && installation.enterprise !== undefined) {
         // support for org wide app installation
         return await database.set(installation.enterprise.id, installation);
-      } else {
+      }
+      if (installation.team.id !== undefined) {
```

Comment:
To make this code compatible with TypeScript, checking the existence of `installation.team` is necessary:

```suggestion
      if (installation.team !== undefined && installation.team.id !== undefined) {
```

If this document is only for TS, the code can be `installation?.team?.id !== undefined`. But this code snippet must be compatible with both JS and TS.

Reply:
Good spot, I've changed it.

I didn't mean to check for `installation.team.id !== undefined` but `installation.team !== undefined`. Which is what I'm currently doing in [the code I'm working on](https://github.com/connorads/lockbot/blob/upgrade-slack-bolt-to-v3/src/handlers/slack/infra.ts#L30).

Don't think we need to check for `installation.team.id !== undefined` if we can trust the type declarations as `id` is not optional.
![image](https://user-images.githubusercontent.com/10026538/120203019-2ece9f80-c21f-11eb-9b5e-bac2cd32a80e.png)

Reply:
Thanks. Looks good to me too 👍